### PR TITLE
core: update 'bump-cli' dependency to introduce Branch input

### PR DESCRIPTION
**Pull request by paulRbr**
Following the latest release of the CLI
https://github.com/bump-sh/cli/releases/tag/v2.4.0 this commit adapts
the GH action to accept a new input parameter `branch` to be able to
deploy versions to dedicated branches.

### CODE REVIEW

File: tests/main.test.ts
Comment (Polo2) [inline tests/main.test.ts:101]: nice refacto 💯

File: tests/setup-test-env.ts
Comment (Polo2) [inline tests/setup-test-env.ts:1]: Nice refacto, this new method `mockEnv`, with `restore()` calls makes specs really easier to read.
We could have imagine here to split this PR in two commits, and keep this change in first commit, to keep changes about branch more isolated... But that's already great, no need to change, your call to decide before merging it

**Review by Polo2 — APPROVED** [review]
So nice to see this new `branch` feature coming to life step by step, well done 👏

File: tests/setup-test-env.ts
Comment (paulRbr) [reply tests/setup-test-env.ts:1]: Actually, I needed to make this change because of interdependent mocked process env value before this PR... I was never resetting the variables and this had wrong values passed from test to test :grimacing:.

This new method was more than necessary :)

**Pull request merged**
